install/0000_90_machine-config-operator_01_prometheus-rules.yaml

```diff
@@ -93,13 +93,62 @@ spec:
             severity: warning
           annotations:
             message: "System memory usage of {{ $value | humanize }} on {{ $labels.node }} exceeds 95% of the reservation. Reserved memory ensures system processes can function even when the node is fully allocated and protects against workload out of memory events impacting the proper functioning of the node. The default reservation is expected to be sufficient for most configurations and should be increased (https://docs.openshift.com/container-platform/latest/nodes/nodes/nodes-nodes-managing.html) when running nodes with high numbers of pods (either due to rate of change or at steady state)."
-    - name: master-nodes-high-memory-usage
+    - name: high-overall-control-plane-memory
```

Comment:
I'd drop `overall-` here, to give space to also focus on a single node at a time, instead of having to always deal with the control-plane in aggregate.

Reply:
I prefer overall - looking at the control plane as a whole makes sense to me. I dont care if a single instance has high mem usage while the services are functioning fine.

Reply:
It shows the descheduler is broken and a burst of load would cause OOM-kills on the mem-loaded node.  Because OOM-kills take out the container, but do not take out the pod, so without functioning descheduling you are unlikely to automatically recover.

Also, there is nothing that requires a `high-control-plane-memory` set to include single-node alerting; it just leaves that option open for the future.  Can we agree that there's no need to close the door on single-node alerting in general, go for the generic name here, and then debate the utility of specific single-node alert rules as they're proposed?

Reply:
> It shows the descheduler is broken and a burst of load would cause OOM-kills on the mem-loaded node. Because OOM-kills take out the container, but do not take out the pod, so without functioning descheduling you are unlikely to automatically recover.

sorry how is the descheduler broken in this context?

>Also, there is nothing that requires a high-control-plane-memory set to include single-node alerting; it just leaves that option open for the future. Can we agree that there's no need to close the door on single-node alerting in general, go for the generic name here, and then debate the utility of specific single-node alert rules as they're proposed?

what ever is less ambiguous - I dont want to figure out if an alert is for a single instance or the entire control plane when dealing with it. 

RE: single node - potentially keep as a warning/info contextual based alert. I care about services atop and their performance.

Reply:
> sorry how is the descheduler broken in this context?

The only recovery mechanism for unbalanced resource consumption is evicting some pods.  The deployment controller (or whoever is creating the pods) will create a replacement pod.  The scheduler will see "whoa, nodeA is slammed; I'll put this on nodeB, where there is lots of space".

The kubelet/kernel on the overloaded node can OOM-kill containers, but they will not evict pods.   The kubelet will just launch a replacement container for the existing pod on the overloaded node.  The descheduler is build to do this sort of pod eviction, but I dunno if we have it wired up to manage our control-plane pods.  And even if we are asking the descheduler to do this today, it doesn't mean it is bug-free and always successful.

> I dont want to figure out if an alert is for a single instance or the entire control plane when dealing with it.

You don't need to figure it out.  The alert will have already done that for you, and can suggest different response plans.  For "only one node is high", you have two options:

* Evict some pods, so the scheduler can move that work elsewhere.
* Scale up the control plane machines.

In the "whole control plane is high" case, only the latter will help.

> RE: single node - potentially keep as a warning/info contextual based alert. I care about services atop and their performance.

I'm ok with `warning` for single node resource alerting.